src/restore.ts

```diff
@@ -35,22 +35,46 @@ async function run(): Promise<void> {
             restoreKeys
         );
 
+        //Check if user wants to save cache despite of failure in any previous job
+        const saveCache = core.getInput(Inputs.SaveOnAnyFailure).toLowerCase();
+        if (saveCache == "true") {
+            core.debug(
+                `Exporting environment variable ${Variables.SaveCacheOnAnyFailure}`
+            );
+            core.exportVariable(Variables.SaveCacheOnAnyFailure, saveCache);
+            core.info(
+                `Input Variable ${Variables.SaveCacheOnAnyFailure} is set to true, the cache will be saved despite of any failure in the build.`
+            );
+        }
+
         if (!cacheKey) {
+            if (core.getInput(Inputs.FailOnCacheMiss).toLowerCase() == "true") {
+                throw new Error(
+                    `Cache with the given input key ${primaryKey} is not found, hence exiting the workflow as the fail-on-cache-miss requirement is not met.`
+                );
+            }
             core.info(
                 `Cache not found for input keys: ${[
                     primaryKey,
                     ...restoreKeys
                 ].join(", ")}`
             );
-
             return;
         }
-
         // Store the matched cache key
         utils.setCacheState(cacheKey);
```

Comment:
Can't put a comment on L25 so putting here. The primary key should be set as output as well. This because user mostly would want to reuse same computed primary key in `save` step rather than recompute it again.